File: netflipster/csharp/Netflix2/repository/UserDao.cs
namespace Netflix2.repository
{
    public class UserDao
    {
        /**
         * Register a user to the system.
         * The user identifier is unique.
         *
         * @param userId identifier of a user
         * @param password password
         * @return true if the user has been successfully registered, false otherwise
         */
        public static bool Register(string userId, string password)
        {
            return DB.Insert($"INSERT INTO USERS(ID, PASSWORD) values('{userId}', '{password}')");
        }

        /**
         * Check if a user
         * @param userId identifier of a user
         * @param password password
         * @return true if the user exists, false otherwise
         */
        public static bool Exist(string userId, string password)
        {
            var numberOfUsers = DB.ExecuteScalar<long>($"SELECT count(1) FROM USERS where id='{userId}' AND password='{password}'");
            return numberOfUsers > 0;
        }
    }
}

File: netflipster/java/src/main/java/repository/UserDao.java
package repository;

import source.DB;

import java.sql.ResultSet;
import java.sql.SQLException;

public class UserDao {

    /**
     * Register a user to the system.
     * The user identifier is unique.
     *
     * @param name     identifier of a user
     * @param password password
     */
    public static void register(String name, String password) {
        DB.insert(String.format("INSERT INTO USERS(NAME, PASSWORD) values('%s', '%s')", name, password));
    }

    /**
     * Check if a user
     *
     * @param name     identifier of a user
     * @param password password
     * @return the user identifier if the user exists; empty string otherwise.
     */
        public static String findId(String name, String password) {
        ResultSet rs = DB.execute(String.format("SELECT ID as total FROM USERS where name='%s' AND password='%s'", name, password));
        try {
            rs.next();
            return String.valueOf(rs.getInt("ID"));
        } catch (SQLException e) {
            e.printStackTrace();
            return "";
        }
    }

}
